### PR TITLE
update hmc830 code. cp to 1.6mA, add docs, other regs to data sheet v…

### PR DESCRIPTION
**Pull request by hartytp**
…alues

### CODE REVIEW

**Comment by hartytp** [comment]
NB once this all works, we should probably set the lock detect window to 1/2 cycles...

File: artiq/firmware/libboard_artiq/hmc830_7043.rs [hunk 1]
```diff
@@ -30,23 +30,22 @@ mod clock_mux {
 mod hmc830 {
     use board::{csr, clock};
 
+    // See "PLLs WITH INTEGRATED VCO - RF APPLICATIONS PRODUCT & OPERATING GUIDE"
```
Comment (sbourdeauducq) [inline artiq/firmware/libboard_artiq/hmc830_7043.rs]:
```
error[E0308]: mismatched types
  --> /home/sb/artiq_drtio/artiq/firmware/libboard_artiq/hmc830_7043.rs:34:44
   |
34 |       const HMC830_WRITES: [(u8, u32); 16] = [
   |  ____________________________________________^
35 | |         (0x0, 0x20), // RESET: software reset
36 | |         (0x0, 0x00), // RESET: normal operation
37 | |         (0x2, 0x01), // REF_DIV: r=1
...  |
48 | |         (0x3, 0x18), // N_DIV: n=24
49 | |     ];
   | |_____^ expected an array with a fixed size of 16 elements, found one with 14 elements
   |
   = note: expected type `[(u8, u32); 16]`
              found type `[(u8, u32); 14]`
```

**Comment by sbourdeauducq** [comment]
Can you also align the comments for readability?

File: artiq/firmware/libboard_artiq/hmc830_7043.rs (hunk 1 above)
Comment (hartytp) [reply artiq/firmware/libboard_artiq/hmc830_7043.rs]: oops, sorry!

File: artiq/firmware/libboard_artiq/hmc830_7043.rs [hunk 2]
```diff
@@ -30,23 +30,22 @@ mod clock_mux {
...
-        (0x9, 0x153fff),
-        (0xa, 0x2046),
-        (0xb, 0x7c061),
-        (0xf, 0x81),
-        (0x3, 0x30), // n_divider
+        (0x0, 0x20), // RESET: software reset
+        (0x0, 0x00), // RESET: normal operation
+        (0x2, 0x01), // REF_DIV: r=1
+        (0x5, 0xe110), // VCO_REG_2: output divider=2, max output gain
+        (0x5, 0x2818), // VCO_REG_3: diff output, auto RFO
```
Comment (jbqubit) [inline artiq/firmware/libboard_artiq/hmc830_7043.rs]: Thanks for the inline bread crumbs @hartytp. :+1

**Pull request closed without merging**